src/lib/pages/contract-details/components/contract-state/index.tsx

```diff
@@ -98,8 +99,22 @@ export const ContractStates = ({ contractAddress }: ContractStatesProps) => {
           isLoading={isFetching || isFetchingNextPage}
           totalData={totalData}
           isCompleted={!hasNextPage}
-          onLoadMore={fetchNextPage}
-          onDownload={handleDownload}
+          onLoadMore={() => {
+            trackContractStatesLoad(AmpEvent.USE_CONTRACT_STATES_LOAD_MORE, {
+              current_states: totalData,
+              namespaces_count: namespaces.length,
+              namespaces,
```

Comment:
Why are you returning `namespaces_count` even though you can obtain the length directly from `namespaces`?